src/adapter/src/catalog.rs

```diff
@@ -6144,6 +6148,16 @@ pub fn is_reserved_name(name: &str) -> bool {
         .any(|prefix| name.starts_with(prefix))
 }
 
+/// Enable catalog features that might be required during planning in
+/// [Catalog::open]. Existing catalog items might have been created while a
+/// specific feature flag turned on, so we need to ensure that this is also the
+/// case during catalog rehydration in order to avoid panics.
+fn enable_features_required_for_catalog_open(session_catalog: &mut ConnCatalog) {
```

Comment:
nit: it would be more idiomatic Rust to add a `with_feature(self, feature: CatalogFeature)` method on `ConnCatalog`, which does what this is doing, enables the feature if it's not already enabled, this way you could use the builder pattern like:

```
let session_catalog = self.for_system_session()
                          .with_feature(CatalogFeature::EnableWithMutuallyRecursive);
```

Reply:
`enable_features_required_for_catalog_open` needs to be called twice, and over time I suspect the number of features that evolve will change (some will be removed and others will be added depending on the current epics and the lifecycle of each feature).

I therefore prefer to have a single method which lives in `catalog.rs` which is where the decision of which of the features need to be enabled for item parsing needs to be made.

I can define it like that:

```rust
fn enable_features_required_for_catalog_open(session_catalog: ConnCatalog) -> ConnCatalog {
    session_catalog
        // enable features required for parsing
        .with_feature(CatalogFeature::EnableWithMutuallyRecursive)
}
```